R/child_care_subsidy.R

```diff
@@ -0,0 +1,125 @@
+                                         0.85 - floor((family_annual_income - 66958)/3000)/100,
+                                         if_else(family_annual_income < 251248,
+                                                 0.5,
+                                                 if_else(family_annual_income < 341248,
+                                                         0.5 - floor((family_annual_income - 251248)/3000)/100,
+                                                         if_else(family_annual_income < 354248,
+                                                                 0.2,
+                                                                 0)))))] 
+  #type of care adjustment
+  .data[, type_of_day_care := if_else(child_age >= 6 & type_of_day_care == "cbdc",
+                                      "oshc",
+                                      if_else(child_age < 6 & type_of_day_care == "oshc",
+                                              "cbdc", 
+                                              type_of_day_care))]
+  
+  hourly_cap <- .data[, if_else(type_of_day_care == "cbdc",
+                                11.77,
+                                if_else(type_of_day_care == "fdc",
+                                        10.90,
+                                        if_else(type_of_day_care == "oshc",
+                                                10.29,
+                                                if_else(type_of_day_care == "ihc",
+                                                        25.48,
+                                                        0))))]
+  
+  annual_cap <- .data[, if_else(family_annual_income > 186958,
+                                10190,
+                                Inf)]
+  
+  activity_test_1 <- .data[, if_else(activity_level < 8,
+                                     0,
+                                     if_else(activity_level <= 16,
+                                             36,
+                                             if_else(activity_level <=  48,
+                                                     72,
+                                                     100)))]
+  activity_test_2 <- .data[, if_else(family_annual_income <= 66985 & activity_level < 8,
+                                     24, 0)]
+  activity_test_3 <- .data[, if_else(early_education_program & type_of_day_care == "cbdc",
+                                     36, 0)]
+  activity_test_4 <- .data[, if_else(activity_exemption, 100, 0)]
+  
+  activity_test <- pmax(activity_test_1, activity_test_2, activity_test_3, activity_test_4)
+  
+  input <- data.table(cost_hour,
+                      hourly_cap,
+                      income_test,
+                      hours_day_care_fortnight,
+                      activity_test,
+                      annual_cap)
+  #calculation
+  output <- input[, pminC(pminC(cost_hour, hourly_cap) * income_test * pminC(hours_day_care_fortnight, activity_test) * 365/14,
```

Comment:
Keep everything inside the data table rather than relying on the parent frame. That is, make sure every variable you're currently creating using `var <- .data[, ...]` are done using `.data[, var := ...]`. This can avoid confusion with variables from the global environment.